File: src/types/conversation.ts
export type AuthorRole = "user" | "assistant" | "system" | "tool" | "unknown";

export interface ConversationSummary {
  id: string;
  title: string;
  createTime: number | null;
  updateTime: number | null;
  messageCount: number;
}

export interface ConversationMessage {
  id: string;
  conversationId: string;
  parentId: string | null;
  authorRole: AuthorRole;
  contentText: string;
  createTime: number | null;
  updateTime: number | null;
  depth: number;
}

export interface ConversationDetail {
  conversation: ConversationSummary;
  messages: ConversationMessage[];
}

File: src-tauri/src/models/conversation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NormalizedConversationRecord {
    pub id: String,
    pub title: String,
    pub create_time: Option<f64>,
    pub update_time: Option<f64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NormalizedConversationPathRecord {
    pub conversation_id: String,
    pub message_id: String,
    pub depth: i64,
    pub is_terminal: bool,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConversationSummary {
    pub id: String,
    pub title: String,
    pub create_time: Option<f64>,
    pub update_time: Option<f64>,
    pub message_count: i64,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConversationMessage {
    pub id: String,
    pub conversation_id: String,
    pub parent_id: Option<String>,
    pub author_role: String,
    pub content_text: String,
    pub create_time: Option<f64>,
    pub update_time: Option<f64>,
    pub depth: i64,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConversationDetail {
    pub conversation: ConversationSummary,
    pub messages: Vec<ConversationMessage>,
}
